lib/SILOptimizer/Transforms/SILMem2Reg.cpp

```diff
@@ -1634,6 +1806,37 @@ void MemoryToRegisters::removeSingleBlockAllocation(AllocStackInst *asi) {
       continue;
     }
 
+    if (auto *sbi = dyn_cast<StoreBorrowInst>(inst)) {
+      if (sbi->getDest() != asi) {
+        continue;
+      }
+      runningVals = {LiveValues::toReplace(asi, /*replacement=*/sbi->getSrc()),
+                     /*isStorageValid=*/true};
+      if (shouldAddLexicalLifetime(asi)) {
+        runningVals = beginLexicalLifetimeAfterStore(asi, inst);
+      }
+      continue;
+    }
+
+    if (auto *ebi = dyn_cast<EndBorrowInst>(inst)) {
+      auto *sbi = dyn_cast<StoreBorrowInst>(ebi->getOperand());
+      if (!sbi) {
+        continue;
+      }
+      if (!runningVals.hasValue()) {
+        continue;
+      }
+      if (sbi->getSrc() != runningVals->value.stored ||
```

Comment:
Could these be asserts or are we able to see an end_borrow for a store_borrow while we have a different non-none value for runningVals?

Reply:
Right, we can have end_borrow for a store_borrow to a different location.